pkg/controller/operands/cdi_test.go

```diff
@@ -1089,14 +1089,14 @@ var _ = Describe("CDI Operand", func() {
 			})
 
 			It("should update if labels are missing", func() {
-				expectedRole := NewConfigReaderRoleForCR(hco, hco.Namespace)
+				expectedRole := NewCdiConfigReaderRole(hco)
 				expectedLabels := expectedRole.Labels
 				expectedRole.Labels = nil
 
 				cl := commonTestUtils.InitClient([]runtime.Object{hco, expectedRole})
 
-				handler := (*genericOperand)(newConfigReaderRoleHandler(cl, commonTestUtils.GetScheme()))
-				res := handler.ensure(req)
+				handler, _ := NewConfigReaderRoleHandler(logger, cl, commonTestUtils.GetScheme(), hco)
+				res := handler[0].ensure(req)
```

Comment:
Please consider the following, to make the test a bit safer:

```suggestion
				handlers, err := NewConfigReaderRoleHandler(logger, cl, commonTestUtils.GetScheme(), hco)
				Expect(res.Err).ToNot(HaveOccurred())
				Expect(handlers).ToNot(HaveLen(1))
				res := handler1[0].ensure(req)
```

Reply:
Ok, but please not the `Expect(res.Err).ToNot(HaveOccurred())` should be checked only after  `res := handler[0].ensure(req)` is evaluated.

Reply:
fixed

Reply:
> Ok, but please note the `Expect(res.Err).ToNot(HaveOccurred())` should be checked only after `res := handler[0].ensure(req)` is evaluated.

typo. should be `Expect(err).ToNot(HaveOccurred())`